plugins/inputs/jenkins/jenkins_test.go

```diff
@@ -0,0 +1,352 @@
+package jenkins
+
+import (
+	"net"
+	"net/http"
+	"net/http/httptest"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/influxdata/telegraf/testutil"
+	"github.com/stretchr/testify/require"
+)
+
+func TestJenkins_Gather(t *testing.T) {
+
+	ts := httptest.NewUnstartedServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		var resp string
+		switch r.RequestURI {
+
+		case "/computer/api/json?depth=1":
+			w.WriteHeader(http.StatusOK)
+			resp = responseNode
+		case "/job/JOB1/api/json":
+			w.WriteHeader(http.StatusOK)
+			resp = responseJob1
+		case "/job/JOB1/1/api/json?depth=1":
+			w.WriteHeader(http.StatusOK)
+			resp = responseJob1Build1
+		case "/job/JOB1/2/api/json?depth=1":
+			w.WriteHeader(http.StatusOK)
+			resp = responseJob1Build2
+		case "/job/JOB1/api/json?tree=allBuilds%5Bnumber%2Curl%5D":
+			w.WriteHeader(http.StatusOK)
+			resp = responseJob1AllBuilds
+		case "/api/json":
+			w.WriteHeader(http.StatusOK)
+			resp = response
+		default:
+			w.WriteHeader(http.StatusNotFound)
+		}
+		w.Write([]byte(resp))
+	}))
+
+	l, _ := net.Listen("tcp", "127.0.0.1:8080")
```

Comment:
I would probably just use httptest.NewServer, but if you listen separately then use port 0 to get a auto assigned port.

Reply:
Ok, I will have a look at that